crates/ruff_linter/src/rules/pep8_naming/rules/invalid_first_argument_name.rs

```diff
@@ -59,7 +60,7 @@ use crate::renamer::Renamer;
 /// [PEP 8]: https://peps.python.org/pep-0008/#function-and-method-arguments
 #[violation]
 pub struct InvalidFirstArgumentNameForMethod {
```

Comment:
I feel conflicted about this change. Or at least, I wouldn't expect methods without arguments to be flagged when reading the documentation:

> Checks for instance methods that use a name other than self for their first argument.

I understand that this implies that the method must have at least one argument for the rule to raise a violation.

Reply:
Yeah, I'm also not sure about this... Do we not have other rules that flag if an instance or class method has zero parameters? If not, maybe we should? I think it's important to flag that, as it's almost certainly a bug, but I'm not sure it's a pep8-naming rule's responsibility to flag it

Reply:
Edit: I think if we change this, then it should probably be a preview only change because it changes the scope of the rule (and it's less clear that it doesn't change the intent of the rule)